# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/clients/manager/ClientSource.cpp
# Commit: df38944a433e
# Full Hash: df38944a433ef38a9bd2795aa1b85c4f49b4185d
# Author: Tim Huang <tihuang@mozilla.com>
# Date: 2020-05-28 21:53:02
# Regressor Bug: 1641153
# File Overlap Count: 1
# Description:
#   Bug 1641153 - Part 3: Revert the assertion back to diagnostic assertion in ClientSource.cpp r=dimi
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D77076
# ==============================================================================

diff -r b0959ede7962 -r df38944a433e dom/clients/manager/ClientSource.cpp
--- a/dom/clients/manager/ClientSource.cpp	Thu May 28 11:39:26 2020 +0000
+++ b/dom/clients/manager/ClientSource.cpp	Thu May 28 11:39:39 2020 +0000
@@ -262,9 +262,10 @@
   // continue to inherit the SW as well.  We need to avoid triggering the
   // assertion in this corner case.
   if (mController.isSome()) {
-    MOZ_ASSERT(spec.LowerCaseEqualsLiteral("about:blank") ||
-               StringBeginsWith(spec, NS_LITERAL_CSTRING("blob:")) ||
-               StorageAllowedForWindow(aInnerWindow) == StorageAccess::eAllow);
+    MOZ_DIAGNOSTIC_ASSERT(spec.LowerCaseEqualsLiteral("about:blank") ||
+                          StringBeginsWith(spec, NS_LITERAL_CSTRING("blob:")) ||
+                          StorageAllowedForWindow(aInnerWindow) ==
+                              StorageAccess::eAllow);
   }
 
   nsPIDOMWindowOuter* outer = aInnerWindow->GetOuterWindow();
